coil/src/androidTest/java/dev/chrisbanes/accompanist/coil/CoilTest.kt

```diff
@@ -175,6 +176,39 @@ class CoilTest {
         loadCompleteSignal.close()
     }
 
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun basicLoad_changeSize() {
+        val loadCompleteSignal = Channel<Unit>(Channel.UNLIMITED)
+        val sizeFlow = MutableStateFlow(128.dp)
+
+        composeTestRule.setContent {
+            val size = sizeFlow.collectAsState()
+            CoilImage(
+                data = resourceUri(R.drawable.blue_rectangle),
+                modifier = Modifier.preferredSize(size.value).testTag(CoilTestTags.Image),
+                onRequestCompleted = { loadCompleteSignal.offer(Unit) }
+            )
+        }
+
+        // Await the first load
+        runBlocking {
+            loadCompleteSignal.receive()
+        }
+
+        // Now change the size
+        sizeFlow.value = 256.dp
+
+        // Await the potential second load (which shouldn't come)
+        runBlocking {
+            val result = withTimeoutOrNull(3000) { loadCompleteSignal.receive() }
```

Comment:
This test will take 3 seconds to run every time... can you decrease this timeout?

Reply:
The problem is that these tests aren't very fast, so it could timeout and be a false positive. 3 seconds is good enough for me, the build is far slower than that anyway 🤣

Reply:
Fair enough 😄